scripts/quests/crystalWar/A_Little_Knowledge.lua

```diff
@@ -73,14 +73,11 @@ quest.sections =
                     -- In addition, need to determine if partial stacks are consumed and implement that if needed.
 
                     if
+                        npcUtil.tradeHas(trade, { { xi.items.ROLANBERRY, 12 } }) and
```

Comment:
This is incorrect.  You can trade less and therefore get less.

Reply:
> This is incorrect. You can trade less and therefore get less.

This directly conflicts with my personal testing as I could NOT trade less than 12.

Reply:
So there is a case that can be updated then if they player trades _less_ than 12, but you're _only accepting_ 12 at a time by this change, which is also incorrect, since they can trade up to 4 stacks and get different results.

Reply:
Though, looking at this, it does check for >= 12, so you can't trade less.  Are you saying that you cannot trade anything that is not a full stack?

Reply:
> Though, looking at this, it does check for >= 12, so you can't trade less. Are you saying that you cannot trade anything that is not a full stack?

You can trade 4 stacks and get 4 vellum. Trade 2 stacks, get 2 vellum. If it's not a full stack, he'll return those rolanberries to you. However, if you get a single piece of vellum during that trade round, you gain that 1 of 3 variables for a chance at the full vellum stack limit.

Reply:
Ok, so it looks like the only thing that needs to change then is not eating the entire container, but instead confirming multiples of stacks

Reply:
> Ok, so it looks like the only thing that needs to change then is not eating the entire container, but instead confirming multiples of stacks

I'm not sure what you mean.

Reply:
```lua
                    if
                        questOption >= 1 and
                        questOption <= 3 and
                        trade:hasItem(xi.items.ROLANBERRY) and
                        trade:getItemCount() >= 12 and
                        trade:hasItemQty(xi.items.ROLANBERRY, trade:getItemCount())
                    then
```
From above, you have to talk to Tucker first (option >= 1), you get 3 changes to trade (<= 3, and unavailable when the value is 4)

Trade can only have Rolanberries, and must have at least 12

```lua
local tuckerEventFinish = function(player, csid, option, npc)
    local numAwarded = quest:getLocalVar(player, 'numAwarded')

    if npcUtil.giveItem(player, { { xi.items.SHEET_OF_VELLUM, numAwarded } }) then
        player:confirmTrade()
        quest:incrementVar(player, 'Option', 1)
    end
end
```

Actually, we're doing this already.  We're only confirming `12 * numAwardedVellum`
```lua
                        local numAwardedVellum = math.min(4, math.floor(trade:getItemCount() / 12))

                        trade:confirmItem(xi.items.ROLANBERRY, numAwardedVellum * 12)
                        quest:setLocalVar(player, 'numAwarded', numAwardedVellum)
```

So the confirm only eats what we reward on.  What is the issue again?

Reply:
Mind you, I get the additional dialogue, and welcome it

Reply:
trade:hasItem does not initiate.

Reply:
Yep, that's the bug.  Needs a second parameter of 12, and then can remove the condition for `trade:getItemCount() >= 12 and`

Reply:
We don't want to use npcUtil with a static value of 12, because we're manually confirming later on in the code

Reply:
> We don't want to use npcUtil with a static value of 12, because we're manually confirming later on in the code

So what should we do here? If I don't confirm the 12 here, then a Option variable would be counted if less than 12 was traded, I would think.

Reply:
I've already set confirming manually instead of relying on the npcUtil function
```lua
                        local numAwardedVellum = math.min(4, math.floor(trade:getItemCount() / 12))

                        trade:confirmItem(xi.items.ROLANBERRY, numAwardedVellum * 12)
                        quest:setLocalVar(player, 'numAwarded', numAwardedVellum)
```

Reply:
You can't enter that condition if less than 12 are traded (at least once the missing parameter is fixed)

Reply:
> I've already set confirming manually instead of relying on the npcUtil function
> 
> ```lua
>                         local numAwardedVellum = math.min(4, math.floor(trade:getItemCount() / 12))
> 
>                         trade:confirmItem(xi.items.ROLANBERRY, numAwardedVellum * 12)
>                         quest:setLocalVar(player, 'numAwarded', numAwardedVellum)
> ```

So, remove the npc_utils line and keep the 

`if
                        questOption >= 1 and
                        questOption <= 3
then`

?

Reply:
```lua
                    if
                        questOption >= 1 and
                        questOption <= 3 and
                        trade:hasItem(xi.items.ROLANBERRY, 12) and
                        trade:hasItemQty(xi.items.ROLANBERRY, trade:getItemCount())
                    then
```

Reply:
Ok, let me test.

Reply:
@claywar 

trade:hasItem refuses to work. Throwing a handling error with nil value.

Reply:
Sheesh, where was my head space that day (and today).  Change hasItem to hasItemQty

Reply:
Yeah, this isn't working at all. NPC is not responding to the trade.

Reply:
Give me a moment.  I'll check locally and get back.  You did talk to Tucker first, right? That's a requirement

Reply:
> Give me a moment. I'll check locally and get back. You did talk to Tucker first, right? That's a requirement

yeah, It's not even giving me the correct dialogue CS neither. I've already resetted my variable. Still no change.

Reply:
```lua
                    if
                        questOption >= 1 and
                        questOption <= 3 and
                        trade:getItemQty(xi.items.ROLANBERRY) >= 12 and
                        trade:hasItemQty(xi.items.ROLANBERRY, trade:getItemCount())
                    then
                    ```

Reply:
Then I'm having an issue getting that variable back to 0.

Reply:
recommiting. Works